### PR TITLE
rds_snapshot module

### PR DESCRIPTION
**Pull request by willthames**
##### SUMMARY
New rds_snapshot module

##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME
rds_snapshot

##### ANSIBLE VERSION
```

ansible 2.6.0 (devel 4e2876be9f) last updated 2018/05/11 13:01:31 (GMT +1000)
   config file = None
   configured module search path = [u'/Users/will/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
   ansible python module location = /Users/will/src/opensource/ansible/lib/ansible
   executable location = /Users/will/src/opensource/ansible/bin/ansible
   python version = 2.7.14 (default, Apr 25 2018, 15:22:05) [GCC 4.2.1 Compatible Apple LLVM 9.1.0 (clang-902.0.39.1)]

```

##### ADDITIONAL INFORMATION
No tests are included with this module, as they are expected to be part of the rds_instance test suite. 

This PR was created from #30746.

### CODE REVIEW

**Comment by mkrizek** [comment]
cc @s-hertel @ryansb

**Comment by willthames** [comment]
WIP until Aurora snapshots are better handled.

File: lib/ansible/modules/cloud/amazon/rds_snapshot.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+EXAMPLES = '''
+# Create snapshot
+- rds_snapshot:
+    instance_name: new-database
+    snapshot: new-database-snapshot
+
+# Delete snapshot
+- rds_snapshot:
+    snapshot: new-database-snapshot
+    state: absent
```
Comment (Akasurde) [inline lib/ansible/modules/cloud/amazon/rds_snapshot.py]:
```suggestion
    state: absent
  register: rds
```

File: lib/ansible/modules/cloud/amazon/rds_snapshot.py [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+- debug:
```
Comment (Akasurde) [inline lib/ansible/modules/cloud/amazon/rds_snapshot.py]: Task seems irrelevant to me.

File: lib/ansible/modules/cloud/amazon/rds_snapshot.py [hunk 3]
```diff
@@ -0,0 +1,181 @@
+#!/usr/bin/python
+# Copyright (c) 2014-2017 Ansible Project
```
Comment (Akasurde) [inline lib/ansible/modules/cloud/amazon/rds_snapshot.py]:
```suggestion
# Copyright (c) 2014 Ansible Project
```

File: lib/ansible/modules/cloud/amazon/rds_snapshot.py [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+---
+module: rds_snapshot
+version_added: "2.9"
+short_description: manage Amazon RDS snapshots
+description:
+     - Creates or deletes RDS snapshots.
+options:
+  state:
+    description:
+      - Specify the desired state of the snapshot
```
Comment (Akasurde) [inline lib/ansible/modules/cloud/amazon/rds_snapshot.py]:
```suggestion
      - Specify the desired state of the snapshot.
```

File: lib/ansible/modules/cloud/amazon/rds_snapshot.py [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+    default: present
+    choices: [ 'present', 'absent']
+  db_snapshot_identifier:
+    description:
+      - The snapshot to manage
```
Comment (Akasurde) [inline lib/ansible/modules/cloud/amazon/rds_snapshot.py]:
```suggestion
      - The snapshot to manage.
```

File: lib/ansible/modules/cloud/amazon/rds_snapshot.py [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+
+# import module snippets
+from ansible.module_utils.aws.core import AnsibleAWSModule
+from ansible.module_utils.aws.rds import get_snapshot, snapshot_to_facts
+
+
+def await_resource(conn, resource_name, status, module):
+    wait_timeout = module.params.get('wait_timeout') + time.time()
+    resource = get_snapshot(conn, resource_name)
+    while not resource or (wait_timeout > time.time() and resource['Status'] != status):
```
Comment (s-hertel) [inline lib/ansible/modules/cloud/amazon/rds_snapshot.py]: There are some built-in waiters that could be used instead. Besides that and the docs suggestions, looks good to me. https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/rds.html#waiters

Aurora still needs some work in general and since the rds module cannot handle aurora for snapshots either yet I don't think it should be a blocker.

Comment (willthames) [reply lib/ansible/modules/cloud/amazon/rds_snapshot.py]: ok, that's good to hear @s-hertel

**Comment by willthames** [comment]
Actions for me to complete in the next six days:
* Improve documentation
* Use boto3 waiters

File: lib/ansible/modules/cloud/amazon/rds_snapshot.py (hunk 2 above)
Comment (willthames) [reply lib/ansible/modules/cloud/amazon/rds_snapshot.py]: Deleted

File: lib/ansible/modules/cloud/amazon/rds_snapshot.py (hunk 5 above)
Comment (willthames) [reply lib/ansible/modules/cloud/amazon/rds_snapshot.py]: Done

File: lib/ansible/modules/cloud/amazon/rds_snapshot.py (hunk 4 above)
Comment (willthames) [reply lib/ansible/modules/cloud/amazon/rds_snapshot.py]: Done

File: lib/ansible/modules/cloud/amazon/rds_snapshot.py (hunk 3 above)
Comment (willthames) [reply lib/ansible/modules/cloud/amazon/rds_snapshot.py]: Done

**Comment by willthames** [comment]
Changed this to WIP until two other PRs are merged (in order)
* mass update to testing policies: https://github.com/ansible/ansible/pull/44387
* fix rds_instance testing policies, tests and bugs (also depends on #44387): TBC

**Review by s-hertel — APPROVED** [review]

**Review by jillr — APPROVED** [review]
